tensorflow/c/kernels.cc

```diff
@@ -279,4 +279,33 @@ TF_Tensor* TF_AllocateOutput(TF_OpKernelContext* context, int index,
     return nullptr;
   }
   return tf_tensor;
-}
\ No newline at end of file
+}
+
+TF_Tensor* TF_AllocateTemp(TF_OpKernelContext* context, TF_DataType dtype, 
+                           int64_t* dims, int num_dims, 
+                           TF_AllocatorAttributes* attributes,
+                           TF_Status* status) {
+  auto* cc_ctx = reinterpret_cast<::tensorflow::OpKernelContext*>(context);
+  TF_SetStatus(status, TF_OK, ""); 
+  tensorflow::gtl::ArraySlice<tensorflow::int64> dimarray(
+      reinterpret_cast<tensorflow::int64*>(dims), num_dims);
+  tensorflow::AllocatorAttributes allocator_attr; 
+  if (attributes->on_host) { 
+    allocator_attr.set_on_host(true); 
+  }
+  tensorflow::Status s;
+  tensorflow::Tensor tensor;
+  TF_Tensor* tf_tensor;
```

Comment:
Comment from @yisitu : Refactor this declaration into line 305, to minimize tf_tensor's lifetime to exactly where it needs to be used.